bl1/aarch32/bl1_exceptions.S

```diff
@@ -73,6 +73,12 @@ debug_loop:
 	ldr	r1, [r8, #(ENTRY_POINT_INFO_PC_OFFSET + 4)]
 	msr	spsr, r1
 
+#ifdef ENTRY_POINT_INFO_LR_OFFSET
```

Comment:
It seems a bit weird that this needs to be programmed to lr_svc whereas lr_mon is being programmed above.

I think this is somthing that can be abstracted within the platform layer. Perhaps the `bl1_plat_prepare_exit()` can do this selectively if the BL32 is optee or Trusty in AArch32 mode? It doesn't have to be in entry_point_info_t in that case.

Reply:
Hi. Sorry for this very late feedback.

`lr_mon` programs the entry point of the next booted image (bl32) when bl1 exits Monitor mode while `lr_svc` is expected by OP-TEE and Trusty (tbc) as the entry point of the non-secure bl33 image. 

When bl1 boots bl32, it needs to set the bl32 entry point AND provide to bl32 the entry point of bl33. In case of bl32=sp_min, the later is available from the structured parameters passed from bl2 to bl1 then to bl32. When bl32 is optee or trusty, these firmware do not handle the structured `bl_params_t`.

Reply:
Ok, I see the issue you are trying to solve. I am thinking of an alternative way to do this:

The ep_info.x3 (or any other unused register) of BL32 can be modified to have the entrypoint information of BL33 in BL2. Then in `bl1_plat_prepare_exit` you can read the x3 from ep_info and write it to `lr_svc` . x3 can be made 0 after this if required.

This way we can avoid changing the the entrypoint_info_t structure for AArch32.

Reply:
Indeed this would work. I am not fund of this since it will prevent bl32 images from benefiting from a 4th native parameters (carried by register `r3`) since it would be used in trusted firmware for the AArch32 specific "bl33 entry point" boot argument.
Note that `entrypoint_info_t` structure is already different between AArch32 and AArch64.

Reply:
Hi @etienne-lms, 
Sorry for the delayed response as I was on holiday.

Hmm,  this would be restricted to Optee in the platform layer and other BL32 images will not be affected (i.e they still would have r0 - r3 available). Ideally Optee should have changed to accept arguments from ARM-TF. Availability of r0-r3 for passing arguments to Optee does not matter because it doesn't consider them anyway. 

We could add more fields to `epinfo` structure but it should be unconditional IMO. But I see this as a Optee specific problem because it doesn't look in `r0-r3` but expects to `lr_svc` to have the address. Hence I would think the solution should be optee specific and within platform code.

Reply:
ok, noticed. Thanks. I think this is better as it leaves x0-x3 to any standardization of optee boot arguments. I will update the p-r to remove the condition.

Reply:
p-r updated: no real change but renaming the `lr` field into `lr_svc`. As before, this field is included in the entry point description structure only for AArch32 mode.